### ISSUE TITLE
Dataset quads and objects don't find the same data

### ISSUE THREAD

**Issue by drewp** [RDFLib/rdflib#2944]
```py
    g = Dataset()
    g.parse(file=Path('fixed_config.n3').open('rt'), format='n3')
    g.parse(file=Path('ui_selections.n3').open('rt'), format='n3')
    g.parse(source='http://10.43.160.4/graph/currentEvents', format='trig')

            for ev in g.quads((cal, EV['event'], None, None)):
                log.info(f'    found an event1 {ev}')
            for ev in g.objects((cal, EV['event'], None, None)):
                log.info(f'    found an event2 {ev}')
```
The first log prints once but the second does not. I expected the queries to match the same statements (returning a quad in  log1 or a the quad's object node in log2).

**Comment by edmondchuc** [comment]
What result do you get if you enable default union on the `Dataset` object?

```python
g = Dataset(default_union=True)
```

**Comment by mbrock** [comment]
I just spent some hours being confused by this. Indeed, enabling `default_union` fixes the problem.

Without that setting, `dataset.triples((s, p, o))` only iterates over the default graph, unlike `.quads((s, p, o, None))`.

That may be intentional, but the documentation for `Dataset` explicitly says that "querying triples return them all regardless of the graph" which as far as I can tell is not true.

**Comment by edmondchuc** [comment]
Yes, the current behaviour of `Dataset.triples` to return only the default graph's triples is quite unintuitive. We plan to change this in v8. See https://github.com/RDFLib/rdflib/discussions/2591#discussioncomment-14724930
